sdk/core/System.ClientModel/src/Internal/ClientModelEventSource.cs

```diff
@@ -0,0 +1,318 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.ClientModel.Primitives;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Diagnostics.Tracing;
+using System.Text;
+
+namespace System.ClientModel.Internal;
+
+[EventSource]
+internal sealed class ClientModelEventSource : EventSource
+{
+    private const int RequestEvent = 1;
```

Comment:
I don't have a good understanding here, but I'm wondering: do these number values need to match the ones in the Azure.Core type?  Is it a problem that we're holding number values that don't correspond (yet) to things we're doing in SCM?

Mostly just looking to build my understanding here, and I can go read some docs as well :).   Implications of my questions might be -- should we add validation, even just with unit tests, that the numbers are the same?  Would it make sense to not yet add values for events SCM can't do, or is there no harm done there?

Reply:
Yeah they do need to match because they are used for filtering events both on the client side like this: https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/samples/Diagnostics.md#applying-filtering-logic and in app insights

Reply:
I don't think it should be a problem to hold values that aren't being used, we'll want to set those aside for future use.

Reply:
The current test suite does check that the numbers are as expected, `RequestEvent` being the expected number for request event:
```csharp
EventWrittenEventArgs args = _listener.SingleEventById(RequestEvent);
Assert.AreEqual(EventLevel.Informational, args.Level);
Assert.AreEqual(SystemClientModelEventSourceName, args.EventSource.Name);
Assert.AreEqual("Request", args.EventName);
```
I think I went through all the numbers and made sure they're all accounted for (minus the ones that aren't being written to yet), but I'll do another pass of that.